src/validation.cpp

```diff
@@ -185,11 +185,67 @@ bool CheckFinalTxAtTip(const CBlockIndex* active_chain_tip, const CTransaction&
     return IsFinalTx(tx, nBlockHeight, nBlockTime);
 }
 
+std::optional<LockPoints> CalculateLockPointsAtTip(CBlockIndex* tip,
+                                                   const CCoinsView& coins_view,
+                                                   const CTransaction& tx)
+{
+    assert(tip);
```

Comment:
To me, this suggests `tip` should be a reference instead of a pointer?

Reply:
`CChain::Tip()` is been passed as an argument, and it could be a `nullptr`.

Reply:
Sorry I should have said the *parameter* `tip`, as in, if a function expects a non-nullptr tip, then perhaps it should take a reference instead of a pointer. But I think I get what you mean, that `CChain::Tip()` returns a pointer and that's passed around pretty frequently in the codebase.